src/main/webapp/app/admin/user-management/user-management.component.ts

```diff
@@ -419,6 +423,42 @@ export class UserManagementComponent implements OnInit, OnDestroy {
         });
     }
 
+    /**
+     * Selects/Unselects all (filtered) users.
+     */
+    selectAllUsers() {
+        if (this.selectedUsers.length === this.users.length) {
+            // Clear all users
+            this.selectedUsers = [];
+        } else {
+            // Add all users
+            this.selectedUsers = [...this.users];
+        }
+    }
+
+    /**
+     * Selects/Unselects all (filtered) users.
+     */
+    selectUser(user: User) {
+        const index = this.selectedUsers.indexOf(user);
+        if (index > -1) {
+            this.selectedUsers.splice(index, 1);
+        } else {
+            // Add all users
+            this.selectedUsers.push(user);
+        }
+    }
+
+    /**
+     * Delete all selected users.
+     */
+    deleteAllSelectedUsers() {
+        for (const user of this.selectedUsers) {
+            this.deleteUser(user.login!, true);
+        }
```

Comment:
The most likely use case for me would be to delete a whole semester worth of students from the system after they no longer are part of any courses. I expect that neither the client nor the server are particularly good at handling a burst of 100s of individual REST requests at the same time.

I think this functionality requires the adaption of the endpoint so that it can receive a list of logins. Then the server can properly batch-process them.

Reply:
I agree :+1: wasn't sure if this would be really a use case, therefore i decided to reuse the existing endpoints. But i will change that accordingly :+1: